### PR TITLE
more strict about spacing and imports

### PR DESCRIPTION
**Pull request by kylpo**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -116,7 +112,8 @@ module.exports = {
     "react/jsx-key": 2,
     "react/jsx-no-bind": [1, {"ignoreRefs": true}],
     "react/jsx-no-duplicate-props": 2,
-    // "react/jsx-no-undef": 2,
+    "react/jsx-no-undef": 2,
```
Comment (kylpo) [inline index.js:115]: will need to be overwritten in our non-new projects

File: index.js [hunk 2]
```diff
@@ -116,7 +112,8 @@ module.exports = {
...
+    "react/react-in-jsx-scope": 2,
```
Comment (kylpo) [inline index.js:116]: will need to be overwritten in our non-new projects

File: index.js [hunk 3]
```diff
@@ -79,18 +79,14 @@ module.exports = {
...
     "quotes": [ 1, "single", {"avoidEscape": true, "allowTemplateLiterals": true} ],
     "semi-spacing": [ 2, { "before": false, "after": true } ],
     "no-multiple-empty-lines": [ 1, { "max": 2 } ],
     "no-extra-semi": 1,
     "space-before-blocks": [ 1, "always" ],
     "space-before-function-paren": [ 1, { "anonymous": "always", "named": "never" } ],
-
-    /* "space-in-parens": [ 2, "always", { "exceptions": ["{}"] } ], */
-
+    "space-in-parens": [ 1, "never" ],
```
Comment (kylpo) [inline index.js]: I've come around to better syntax coloring of parens being better than padding with spaces.

File: index.js [hunk 4]
```diff
@@ -79,18 +79,14 @@ module.exports = {
     "no-trailing-spaces": 1,
     "no-useless-constructor": 1,
     "one-var": [ 1, { "uninitialized": "always", "initialized": "never" } ],
-
-    /* "object-curly-spacing": [ 2, "never" ], */
-
+    "object-curly-spacing": [ 1, "never" ],
```
Comment (kylpo) [inline index.js]: was always the style, just not possible to enforce with the previous paren spacing rule

File: index.js [hunk 5]
```diff
@@ -20,8 +20,8 @@ module.exports = {
   "rules": {
     "no-unexpected-multiline": 1,
     "semi": [1, "never"],
-
     "comma-dangle": [ 1, "always-multiline" ],
+    "no-case-declarations": 1,
```
Comment (kylpo) [inline index.js:24]: helps avoid the const/let hoisting issue of `case` statements. See the "OR" part of https://github.com/kylpo/react-playbook/blob/master/best-practices/redux.md#reducers for more

File: index.js (hunk 4 above)
Comment (mosesoak) [inline index.js]: could the setting just be 0?

Comment (mosesoak) [reply index.js]: I dunno, these types of rules feel constricting. I've always been of the camp that developers should be able to do what feels right in context when it comes to minor formatting. I think the fact that you used to like it one way, then changed your tune is a good case in point, that it's better to just leave more things lenient and be able to get a feel for what you like.

Comment (mosesoak) [reply index.js]: (in regards to the parens thing and this)

Comment (kylpo) [reply index.js]: Thanks for the comments. Yes it is more constricting, but isn't that the point of eslint?

For curly-spacing: this is a _semantic_ style decision that I think we've already decided on. When skimming, I can tell that it is an object and not a destructured one. If there is padding between the curlies, I know it must be destructured.

For parens: using our editors to better color parens is just a more pragmatic solution. I still prefer the looks of spacing within parens, but when hacking away, I'm too lazy to press the space bar twice. Someday I hope editors will allow me to specify rules to auto-pad parens in some situations and not in others, but we aren't there yet. Maybe https://github.com/jlongster/prettier will become configurable-enough for us to use soon.

It is important to remember that the point of these rules is to better enable us to write code that 1) We don't hate looking at, and 2) We are better able to skim.

**Pull request merged**